source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -52,14 +52,19 @@ void WatcherImpl::addWatch(const std::string& path, uint32_t events, OnChangedCb
 
 void WatcherImpl::onInotifyEvent() {
   while (true) {
-    char buffer[sizeof(inotify_event) + NAME_MAX + 1];
+    uint8_t buffer[sizeof(inotify_event) + NAME_MAX + 1];
     ssize_t rc = read(inotify_fd_, &buffer, sizeof(buffer));
     if (rc == -1 && errno == EAGAIN) {
       return;
     }
+    ASSERT(rc >= 0);
+    if (rc < 0) {
```

Comment:
Sorry to be pedantic here, but we should either cover this with `RELEASE_ASSERT` or we should test this case which would involve using the syscall mocks (or could wait for @alyssawilk injectable stuff to land in the other PR). I'm fine with either.

Reply:
I'm willing to use `RELEASE_ASSERT`, but how is that better than `ASSERT` here? An unexpected return value from inotify doesn't seem to justify crashing Envoy running in production mode.

Reply:
I think the idea is that this is so unusual, e.g. returning -2, that something Real Bad must have happened (e.g. memory corruption) and there is no sensible way to make forward progress.

Reply:
Right, see the error handling section in the style guide. IMO this is an acceptable place to just crash. If you don't want to just crash we need to cover the error handling.

Reply:
OK, changed to `RELEASE_ASSERT`.